### PR TITLE
ci: Add ABI check job

### PR DESCRIPTION
**Pull request by hadess**

### CODE REVIEW

**Comment by hadess** [comment]
Are the jobs in the `build.yml` file not run? This is my first time modifying a github workflows definition...

**Comment by hadess** [comment]
> Are the jobs in the `build.yml` file not run? This is my first time modifying a github workflows definition...

All sorted (the jobs don't appear straight away in the list of jobs to check, so I ended up with 3/3 from the Flatpak job showing at first)

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -88,3 +88,23 @@ jobs:
           publish_dir: ./_build/doc/libportal-1/
           cname: libportal.org
           destination_dir: ./
+
+  fedora-35-abi:
+    name: Fedora 35 ABI check
```
Comment (GeorgesStavracas) [inline .github/workflows/build.yml:93]: Will the ABI checker run differently depending on the distro version? If not, I'd recommend calling it `ABI check` only

**Review by GeorgesStavracas — COMMENTED** [review]
One minor comment, looks good otherwise

**Review by GeorgesStavracas — APPROVED** [review]

**Pull request merged**